kolibri/core/content/utils/transfer.py

```diff
@@ -184,16 +197,77 @@ def __iter__(self):
         return self
 
     def next(self):
+        if self.cancel_check():
+            self._kill_gracefully()
+
         try:
-            return super(FileDownload, self).next()
-        except ConnectionError as e:
+            chunk = super(FileDownload, self).next()
+            self.transferred_size = self.transferred_size + self.block_size
+            return chunk
+        except Exception as e:
+            retry = retry_import(e)
+            if not retry:
+                raise
+
             logger.error("Error reading download stream: {}".format(e))
-            raise
+            self.resume()
+            return self.next()
 
     def close(self):
-        self.response.close()
+        if hasattr(self, "response"):
+            self.response.close()
         super(FileDownload, self).close()
 
+    def resume(self):
+        if self.cancel_check():
+            return
+        try:
+            logger.info(
+                "Waiting for 30 seconds before retrying import: {}\n".format(
+                    self.source
+                )
+            )
+            sleep(30)
+
+            byte_range_resume = None
+            # When internet connection is lost at the beginning of start(),
+            # self.response does not get an assigned value
+            if hasattr(self, "response"):
+                # Use Content-Length header to check if range requests are supported
+                # For example, range requests are not supported on compressed files
+                byte_range_resume = self.response.headers.get("content-length", None)
```

Comment:
Isn't the standard server header `accept-ranges: bytes`? There could be other proxies or server setups that prevent byte range requests, not sure which header is a more reliable indicator -- but per the spec, I'm assuming this would be.

Reply:
ah I was using the header `accept-ranges: bytes` previously, but it didn't work well when the files are the ones we compressed on google cloud storage. In this case, somehow cloudflare thinks it's accepting range requests and sends the header `accept-ranges: bytes`. But when I set the header with `{"Range": "bytes={}-"}`, it errors out by saying the link does not support range request. So instead I used `content-length` because it seems like gzip files do not support range requests and thus do not have the `content-length` header. 
Or maybe I can set both conditions in this case?

Reply:
Ok, makes sense. Sounds like they're not respecting the standard, so what you're doing is safer.

>Or maybe I can set both conditions in this case?
Yeah, that could be good. Only try to resume if both headers are set.

Reply:
thank you! I will update the code with that :)